controllers/playlists.controller.js

```diff
@@ -0,0 +1,207 @@
+const { Playlist } = require("../models/playlist.model");
+const { extend } = require("lodash");
+
+const createPlaylist = async (req, res) => {
+	try {
+		let playlistDetails = req.body;
+		const userId = req.user._id;
+		const NewPlaylist = new Playlist({ userId, ...playlistDetails });
+		console.log({ playlistDetails, NewPlaylist });
+		let savedNewPlaylist = await NewPlaylist.save();
+
+		savedNewPlaylist = await savedNewPlaylist
+			.populate({ path: "videos.video" })
+			.execPopulate();
+		savedNewPlaylist.userId = null;
+		res.status(201).json({ response: savedNewPlaylist });
+	} catch (error) {
```

Comment:
you can write a asyncErrorHandler wrapper to avoid writing try catch everytime. You can contact me if you want help with this. :)